freemarker-wrapper/src/main/java/io/codeontap/freemarkerwrapper/files/processors/LayerProcessor.java

```diff
@@ -23,24 +26,53 @@
     protected Map<String, String> fileSystem;
     protected Map<String, Layer> layerMap = new LinkedHashMap<>();
     private Map<String, List<Path>> filesPerLayer = new LinkedHashMap<>();
+    protected Configuration configuration;
 
     public abstract void createLayerFileSystem(LayerMeta meta) throws RunFreeMarkerException;
+    public abstract void postProcessMeta(LayerMeta meta);
     public abstract JsonObjectBuilder buildInformation(JsonObjectBuilder jsonObjectBuilder, Layer layer, Path file);
     public abstract JsonArrayBuilder buildLayers(LayerMeta meta);
 
+    public void setConfiguration(Configuration configuration) {
+        this.configuration = configuration;
+    }
+
     public JsonArray getLayers(LayerMeta meta) throws RunFreeMarkerException {
         createLayerFileSystem(meta);
+        postProcessMeta(meta);
         return buildLayers(meta).build();
     }
 
+    public void initLayers(LayerMeta meta) throws RunFreeMarkerException {
+        if(meta.getStartingPath()!=null && !meta.getStartingPath().startsWith("/")){
+            meta.setStartingPath("/".concat(meta.getStartingPath()));
+        }
+        createLayerFileSystem(meta);
+    }
+
     public Set<JsonObject> getLayerTree(LayerMeta meta) throws RunFreeMarkerException{
         Set<JsonObject> output = new LinkedHashSet<>();
 
         if(meta.getStartingPath()!=null && !meta.getStartingPath().startsWith("/")){
             meta.setStartingPath("/".concat(meta.getStartingPath()));
         }
 
+        if (configuration.getSharedVariableNames().contains("fileSystem")){
+            fileSystem = (TreeMap)((DefaultMapAdapter)configuration.getSharedVariable("fileSystem")).getWrappedObject();
+        }
+
+        if (configuration.getSharedVariableNames().contains("layerMap")){
+            layerMap = (LinkedHashMap)((DefaultMapAdapter)configuration.getSharedVariable("layerMap")).getWrappedObject();
+        }
+
+        if(fileSystem == null){
+            createLayerFileSystem(meta);
```

Comment:
@ml019 , please note that you don't have to call initialiseCMDBFileSystem()/initialisePluginFilesystem() explicitly 
In this case the file system will be initialised when the first search is run, but can be re-init anytime you call initialiseCMDBFileSystem()/initialisePluginFilesystem()